### ISSUE TITLE
Suggestion: AUR package

### ISSUE THREAD

**Issue by margual56** [Slimbook-Team/slimbook_service#8]
## Fluff
Hi! Firstly, thank you for your amazing work with all your software and your Pro X laptop, I'm loving it!

## TL;DR
However, I would like to be able to install this on Arch. I have taken the liberty to write a basic PKGBUILD file to save you some work:
```bash
# Maintainer:
# Contributor: margual56

_pkgname=slimbook_service
pkgname=$_pkgname-bin
pkgver=0.3.1
pkgrel=0
pkgdesc='Service to get desktop notifications from your device'
url='https://github.com/Slimbook-Team/slimbook_service'
license=('GPL3')
arch=('x86_64')
depends=('python>=3')
makedepends=('python-pip')
provides=($_pkgname)
conflicts=($_pkgname)

source=("https://github.com/Slimbook-Team/slimbook_service/releases/download/$pkgver/slimbook_$pkgver.tar.xz")

md5sums=('SKIP')

prepare() {
	tar -xf slimbook_$pkgver.tar.xz
}

build() {
	pip install zmq evdev
}

package() {
	install -Dm755 "${srcdir}/slimbook-${pkgver}/bin/slimbookindicator" "${pkgdir}/usr/bin/slimbookindicator"
	cp -r "${srcdir}/slimbook-${pkgver}/etc/" "${pkgdir}/"
	cp -r "${srcdir}/slimbook-${pkgver}/usr/" "${pkgdir}/"
	install -Dm644 "${srcdir}/slimbook-${pkgver}/usr/share/slimbook/icons/slimbook_be1ofus_dark.svg" "${pkgdir}/usr/share/pixmaps/slimbook_be1ofus_dark.svg"
	install -Dm644 "${srcdir}/slimbook-${pkgver}/usr/share/slimbook/icons/slimbook_be1ofus_light.svg" "${pkgdir}/usr/share/pixmaps/slimbook_be1ofus_light.svg"
	mkdir -p "${pkgdir}/usr/share/applications/"
	install -Dm644 "${srcdir}/slimbook-${pkgver}/usr/share/slimbook/slimbook-client-autostart.desktop" "${pkgdir}/usr/share/applications/slimbook-client-autostart.desktop"
	install -Dm644 "${srcdir}/slimbook-${pkgver}/usr/share/slimbook/slimbook-indicator.desktop" "${pkgdir}/usr/share/applications/slimbook-indicator.desktop"
}

```

It works on my machine 👍🏻 

## Improvements
- I have not added an instruction to enable the `slimbook.service`.
- The `build()` step could be improved by adding a `requirements.txt` to the repository and installing all dependencies.
- There are currently two `.desktop` files being added. One of them should be removed, but I'm not sure which one to remove from the PKGBUILD

---

I would love it if you guys tested it and uploaded it to the AUR. Thank you for your work!

**Comment by MarSlimbook** [comment]
Wow, looks great!
One of the desktop files if for auto starting the program, the other to launch it from the applications menu, they could be simplified to one file maybe.
The packages are uploaded to Manjaro, but I can't find the PKGBUILD, however dependencies are visible, and it depends of another of our packages (slimbook-qc71-dkms) to show notifications of some fn+fx hotkeys, so I'll take a look on how to upload PKGBUILDS for arch users.
Thank you!

**Comment by margual56** [comment]
Thank you for reviewing it 😄

The package is not present in the AUR ([https://aur.archlinux.org/packages?O=0&K=slimbook](https://aur.archlinux.org/packages?O=0&K=slimbook)) 

The process to upload it to the AUR is "simple", you just need to [create an account on the AUR](https://wiki.archlinux.org/index.php?title=Special:CreateAccount) (although I suppose you already have one), then add an ssh key to it.

After that, you clone the AUR repository with the name of the package you want to create (replace "pkgbase" in the following url): `git clone ssh://aur@aur.archlinux.org/pkgbase.git`

Then, you need to upload the PKGBUILD and create the .SCRINFO:
```bash
$ makepkg --printsrcinfo > .SRCINFO
$ git add PKGBUILD .SRCINFO
$ git commit -m "useful commit message"
$ git push
```

Done!

Source: [https://wiki.archlinux.org/title/AUR_submission_guidelines](https://wiki.archlinux.org/title/AUR_submission_guidelines)

**Comment by phnx47** [comment]
@margual56 Hey, maybe will you publish yourself to AUR? Then you can transfer package to Slimbook when they ready or be as Co-Maintainer.

**Comment by margual56** [comment]
Sure thing 👍 

I have a busy day today, but I hope by tomorrow or the day after I can have it done.

I do need the usernames of the co-maintainers you want me to add.

Cheers!

**Comment by phnx47** [comment]
@margual56 I'm not from Slimbook :) I just use Slimbook laptop and maintain some packages in AUR. I wish improve ArchLinux support for Slimbook. Wait answer from @MarSlimbook. I can help with tests and support packages as well :)

**Comment by MarSlimbook** [comment]
Hi, @margual56 here is a [PKGBUILD example](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=slimbookamdcontroller), there we are as co-mantainers. 
I apologize for not being more active towards this, and thank you so much for offering to do this.
PD: it will upoad an update soon.

**Comment by MarSlimbook** [comment]
Well, it might be closed if you don't need F2 F5 notifications. I might try to add slimbook-qc71-dkms as a dependency later.

**Comment by margual56** [comment]
Hi @MarSlimbook, yes I am the proud owner of a ProX 14 :)

I am getting those notifications just fine, although they don't tell you if the function is activated or not:

![Super Key Lock state changed](https://user-images.githubusercontent.com/30444886/202533714-23521b16-9dc5-45fa-adf9-59277a997d1c.png)
![Silent Mode state changed](https://user-images.githubusercontent.com/30444886/202533778-be86b9e3-32f2-4232-9209-4fe43169380f.png)

Also, I can't seem to find the package `slimbook-qc71-dkms` in the AUR. Is it needed for something?

Finally, I am getting some suggestions for the PKGBUILD of `slimbookservice`, so I may end up being quite better than the one I initially suggested.

**Comment by MarSlimbook** [comment]
slimbook-qc71-dkms is a module that allows the service to check the state of the keys.

**Comment by j3j5** [comment]
Hey, very friendly reminder in case you can get the module `slimbook-qc71-dkms` uploaded. 
I'm also a very proud owner of a Prox14 (KDE version, actually) and was very happy to find out some of the Slimbook apps are on the AUR already. Keep it up!

**Comment by j3j5** [comment]
So, quick update, I followed the instructions [on here](https://github.com/Slimbook-Team/slimbook-qc71-dkms/blob/master/slimbook-qc71/README.md) and was able to get the qc71 kernel module up and running on Arch, so now F2 and F5 do what they're supposed to do (super key toggle and performance mode toggle). I may give it a chance to create a PKGBUILD, if I do, I'll ping it here so @margual56 can add it as a dependency of `slimbook-service-bin`. 
By the way @MarSlimbook , is it documented anywhere what the performance mode is supposed to do? Is it related to TLP (like `slimbook-battery`) or not at all? Thanks!

**Comment by MarSlimbook** [comment]
If you are in a Pro-X, there are two performance modes, silent and balanced (as in the BIOS), so the first one reduces your cpu TDP  and fans scaling profile to avoid noise during workloads.

**Comment by margual56** [comment]
Closing because the package is created and can be found in [https://aur.archlinux.org/packages/slimbookservice-bin](https://aur.archlinux.org/packages/slimbookservice-bin)
